File: Java/프로그래머스/3/43105. 정수 삼각형/정수 삼각형.java
import java.util.*;

class Solution {
    public int solution(int[][] triangle) {
        int answer = 0;
        int[][] arr = new int[triangle.length][triangle.length];
        arr[0][0] = triangle[0][0];

        for (int i = 1; i < triangle.length; i++) {
            arr[i][0] = arr[i - 1][0] + triangle[i][0];

            for (int j = 1; j <= i; j++) {
                arr[i][j] = Math.max(arr[i - 1][j - 1], arr[i - 1][j]) + triangle[i][j];
            }
        }

        int[] max = arr[arr.length - 1];
        Arrays.sort(max);
        answer = max[max.length - 1];
        return answer;
    }
}

File: Python/프로그래머스/3/43105. 정수 삼각형/정수 삼각형.py
def solution(triangle):
    n = len(triangle)
    arr = [[0] * n for _ in range(n)]
    arr[0][0] = triangle[0][0]
    for i in range(1, n):
        arr[i][0] = arr[i - 1][0] + triangle[i][0]

        for j in range(1, i + 1):
            arr[i][j] = max(arr[i - 1][j - 1], arr[i - 1][j]) + triangle[i][j]

    return max(arr[len(arr) - 1])